cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -835,8 +840,12 @@ const FunctionDoc subsecond_doc{
 const FunctionDoc strftime_doc{
     "Format timestamps according to a format string",
     ("For each input timestamp, emit a formatted string.\n"
-     "The time format string and locale can be set using StrftimeOptions.\n"
-     "An error is returned if the timestamps don't have a defined timezone,\n"
+     "The time format string and locale can be set using StrftimeOptions. "
+     "Output precision of %S (seconds) flag depends on the input timestamp precision. "
+     "Timestamps with second precision are represented as integers while milliseconds, "
+     "microsecond and nanoseconds are represented as fixed floating point numbers with "
+     "3, 6 and 9 decimal places respectively.\n"
```

Comment:
I would also add something about "To obtain integer seconds, cast to timestamp with second resolution" to be explicit about the workaround.

Reply:
Done.